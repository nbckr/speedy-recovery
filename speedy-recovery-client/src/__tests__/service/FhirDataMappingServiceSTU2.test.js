/*
* Speedy Recovery -- A patient-centred app based on the FHIR standard facilitating communication between paediatric
* patients, parents and hospital staff
*
* Copyright (C) 2019 University College London
*
* This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General
* Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option)
* any later version.
* This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
* warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
* details.
* You should have received a copy of the GNU Affero General Public License along with this program. If not,
* see http://www.gnu.org/license/.
* */

/* This file tests the mapping from FHIR DSTU2 resources to our internal format*/

import {
  fhirMapAppointment,
  fhirMapCarePlan,
  fhirMapCondition,
  fhirMapMedication,
  fhirMapMedicationDispense,
  fhirMapPerson,
  fhirMapPractitioner
} from "../../service/FhirDataMappingService";
import fhirExamplePatient from "../test_input/fhir_resources_stu2/FhirExamplePatientSTU2.json";
import fhirShortApp from "../test_input/fhir_resources_stu2/FhirExampleAppointmentSTU2.json";
import fhirExampleCondition from "../test_input/fhir_resources_stu2/FhirExampleConditionSTU2.json";
import fhirExampleMedication from "../test_input/fhir_resources_stu2/FhirExampleMedicationSTU2.json";
import fhirExampleCarePlan from "../test_input/fhir_resources_stu2/FhirExampleCarePlanSTU2.json";
import fhirExampleMedicationDispense from "../test_input/fhir_resources_stu2/FhirExampleMedicationDispenseSTU2.json";
import fhirExamplePractitioner from "../test_input/fhir_resources_stu2/FhirExamplePractitionerSTU2.json";

test("map FHIR Patient resource to internal user data model", () => {
  const user = fhirMapPerson(fhirExamplePatient, "2");
  expect(user).toMatchSnapshot();
});

test("map FHIR Appointment resource to calendar data model", () => {
  const appointment = fhirMapAppointment(fhirShortApp, "2");
  expect(appointment).toMatchSnapshot();
});

test("map FHIR Condition resource to internal data model", () => {
  const condition = fhirMapCondition(fhirExampleCondition, "2");
  expect(condition).toMatchSnapshot();
});

test("map FHIR Medication resource to internal data model", () => {
  const medication = fhirMapMedication(fhirExampleMedication, "2");
  expect(medication).toMatchSnapshot();
});

test("map FHIR CarePlan resource to internal data model", () => {
  const carePlan = fhirMapCarePlan(fhirExampleCarePlan, "2");
  expect(carePlan).toMatchSnapshot();
});

test("map FHIR MedicationDispense resource to internal data model", () => {
  const medicationDispense = fhirMapMedicationDispense(
    fhirExampleMedicationDispense,
    "2"
  );
  expect(medicationDispense).toMatchSnapshot();
});
test("map FHIR Practitioner resource to internal data model", () => {
  const practitioner = fhirMapPractitioner(fhirExamplePractitioner, "2");
  expect(practitioner).toMatchSnapshot();
});
