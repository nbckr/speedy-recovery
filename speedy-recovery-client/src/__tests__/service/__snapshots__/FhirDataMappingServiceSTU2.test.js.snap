// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map FHIR Appointment resource to calendar data model 1`] = `
Object {
  "appType": "Undefined in STU2",
  "comment": "Further expand on the results of the MRI and determine the next actions that may be appropriate.",
  "created": "Undefined in STU2",
  "description": "Discussion on the results of your recent MRI",
  "end": 2013-12-10T11:00:00.000Z,
  "id": "example",
  "indication": "Undefined in STU2",
  "location": "South Wing, second floor",
  "patient": "Peter James Chalmers",
  "practitioner": "Dr Adam Careful",
  "priority": 5,
  "start": 2013-12-10T09:00:00.000Z,
  "status": "booked",
  "supportingInfo": "Undefined in STU2",
  "title": "Brian MRI results discussion",
}
`;

exports[`map FHIR Condition resource to internal data model 1`] = `
Object {
  "bodySite": "Left Ear",
  "clinicalStatus": "Undefined in STU2",
  "onsetDateTime": 2012-05-24T00:00:00.000Z,
  "severity": "Severe",
  "summary": "Burnt Ear",
  "verificationStatus": "confirmed",
}
`;

exports[`map FHIR Medication resource to internal data model 1`] = `
Object {
  "content": "Undefined in STU2",
  "form": "Capsule",
  "id": "medexample006",
  "imageURL": "Undefined in STU2",
  "isBrand": true,
  "isOverTheCounter": "Undefined in STU2",
  "name": "Apo-Amoxi Cap 250mg",
  "producer": "Undefined in STU2",
}
`;

exports[`map FHIR Patient resource to internal user data model 1`] = `
Object {
  "address": "534 Erewhon St, PleasantVille, Rainbow, Vic, 3999",
  "birthDate": "1974-12-25",
  "careProvider": "Undefined in STU2",
  "email": "Undefined in STU2",
  "gender": "male",
  "name": "Peter James Chalmers",
  "phone": "(03) 5555 6473",
  "role": "Patient",
}
`;
